packages/react-uploader/src/providers/Uploader.tsx

```diff
@@ -117,6 +121,36 @@ export function UploaderProvider ({
       })
       setUploadProgress({})
       return result
+    },
+    async uploadCAR (car: Blob) {
+      if (space == null) throw new Error('missing space')
+      if (agent == null) throw new Error('missing agent')
+
+      const storedShards: CARMetadata[] = []
+      setStoredDAGShards(storedShards)
+
+      const conf = {
+        issuer: agent,
+        with: space.did(),
+        audience: servicePrincipal,
+        proofs: await getProofs([
```

Comment:
If I recall correctly you don't actually need to include these proofs manually - `access-client` should look through its list of proofs and include relevant proofs automatically. Could be misremembering so worth testing....

Reply:
Hmm, I thought that was `w3up-client`? Remember we're using `upload-client` here...

Reply:
oh right - what I'm thinking of is actually in `agent.js` in `access-client` which is I think what `getProofs` ends up calling into - what you've got is exactly right!

this is probably another argument in favor of (refactoring this to use `w3up-client` rather than using the upload and access clients separately...)[https://github.com/web3-storage/w3ui/issues/295]